docs/trex_getdata.md

```diff
@@ -133,6 +177,51 @@ The following example returns data for the first logical table that is used by a
  });
 
 ```
+
+You can specify the number of rows of data to return by setting `GetUnderlyingDataOptions.maxRows` property. If unspecified (`maxRows == '0'`), the call to `getUnderlyingTableDataAsync` requests all rows in the logical table. Note that the maximum number of rows returned from the `getUnderlyingTableDataAsync` method is limited to 10,000 rows. You can use the `DataTable` property, `isTotalRowCountLimited`, to test whether there is more data. A value of true indicates that the calling function requested more rows than the limit (10,000) and the underlying data source contains more rows than can be returned. If the data contains more than 10,000 rows, use `getUnderlyingTableDataReaderAsync` method instead.
+
+
+#### 2b. Get data from the logical table(s) (more than 10,000 rows)
+
+Starting with Tableau 2022.4 and the Dashboard Extensions API library v1.10, if the data contains more than 10,000 rows, use the `getUnderlyingTableDataReaderAsync` method to get data from a worksheet. In this case, you create a `DataTableReader` to iterate through the pages of data. You process each `DataTable` sequentially. You can control the page size, using the optional `pageRowCount` parameter when you call `getUnderlyingTableDataReaderAsync`. The default page size is 10,000 rows.
+
+The basic steps are as follows:
+
+1. Using the `logicalTable.id`, call the `getUnderlyingTableDataReaderAsync` method to create the `DataTableReader`.
+
+1. Create a loop to retrieve each page of underlying data, using `DataTableReader.pageCount` or `DataTableReader.totalRowCount` property of the `DataTableReader` to determine the number of pages to process.
+
+1. Use the `getPageAsync` method to get the DataTable from each page.
+
+1. After you have retrieved all pages of the summary data, call the `releaseAsync()` method to free up memory from the `DataTableReader`.
+
+```javascript
+
+// assumes this code is in an async method
+
+const worksheet = tableau.extensions.dashboardContent.dashboard.worksheets.find(w => w.name === "Sale Map");
+// Call to get the underlying logical tables used by the worksheet
+const underlyingTablesData = await worksheet.getUnderlyingTablesAsync();
+const logicalTableId = underlyingTablesData[0].id;
+// Use the above logicalTableId to get the underlying data reader on the active sheet
+const dataTableReader = await worksheet.getUnderlyingTableDataReaderAsync(logicalTableId);
+// loop through the pages
+const pageNumber = 0;
+while (pageNumber < dataTableReader.pageCount) {
```

Comment:
Change from while to for.
for (let pageNumber = 0; pageNumber < dataTableReader.pageCount; pageNumber++)